src/main/java/org/jabref/logic/externalfiles/LinkedFileHandler.java

```diff
@@ -59,6 +60,10 @@ public boolean moveToDefaultDirectory() throws IOException {
                     filePreferences.getFileDirectoryPattern());
         }
 
+        if (filePreferences.getFileDirectory().isPresent()) {
+            targetDirectoryName = filePreferences.getFileDirectory().get().toString() + File.separator + targetDirectoryName;
```

Comment:
Please use Java 8 API to construct 

```suggestion
            targetDirectoryName = filePreferences.getFileDirectory().get().resolve(targetDirectoryName).toString();
```